llms/mlx_lm/utils.py

```diff
@@ -206,15 +210,18 @@ def _step(y):
         if repetition_context_size:
             if len(repetition_context) > repetition_context_size:
                 repetition_context = repetition_context[-repetition_context_size:]
-        return y, prob
+        return y, prob, logits
```

Comment:
I would simplify this a bit:
1. Always just return `y, log_probs` and get rid of `prob` and get rid of the kwarg `return_step_logits`
2. Fix this to return the `log_probs` and not the `logits` which I think is wrong
3. Update the generate function to compute the exp of the `log_probs[t]` when it needs the actual token prob

Reply:
> I would simplify this a bit:
> 
> 1. Always just return `y, log_probs` and get rid of `prob` and get rid of the kwarg `return_step_logits`
> 2. Fix this to return the `log_probs` and not the `logits` which I think is wrong
> 3. Update the generate function to compute the exp of the `log_probs[t]` when it needs the actual token prob

@awni The OpenAI completions response expects the log probability for each returned token as well as the top N most likely tokens at each token sequence position and their corresponding log probabilities.  How can those be calculated without the logits at each generation step?

Reply:
> log probability for each returned token as well as the top N most likely tokens

You can get both of those from the `log_probs`. By `log_probs` I mean the vector of log probabilities for every token.

Reply:
@awni When you say **log_probs**, which variable are you referring to in that __step_ function? There is only the **logits** variable, which has the vector of raw, conditional probabilities for every token in the vocabulary and **prob**, which results from a call to the _sample_ function.  The _sample_ function returns the softmax probability for only the sampled token, losing the probabilities for the other tokens.

Without the probabilities for the other tokens in the vocabulary, how can you calculate the top N most likely ones?